### PR TITLE
Allow restricting date field date to past or future in forms

### PR DESCRIPTION
**Pull request by ossilva**

### CODE REVIEW

File: resources/translations/en.edn [hunk 1]
```diff
@@ -300,6 +300,9 @@
                 :column-key "ID"
                 :column-label "Label"
                 :confirm-remove-field "Remove this field?"
+                :date-bound {:future "Future"
+                              :none "Not limited"
+                              :past "Past"}
```
Comment (Macroz) [inline resources/translations/en.edn:305]: The indentation looks a bit off here? One space too much?

File: resources/translations/fi.edn [hunk 2]
```diff
@@ -278,6 +278,9 @@
                 :column-key "Tunniste"
                 :column-label "Otsikko"
                 :confirm-remove-field "Poista kenttä?"
+                :date-bound {:none "Ei rajausta"
+                              :past "Menneet"
+                              :future "Tulevat"}
```
Comment (Macroz) [inline resources/translations/fi.edn:283]: Same indentation problem here. Perhaps `cljfmt` is not run to `edn` files at the moment.

File: resources/translations/sv.edn [hunk 3]
```diff
@@ -278,6 +278,9 @@
                 :column-key "Id"
                 :column-label "Titel"
                 :confirm-remove-field "Ta bort fält?" ; TODO check
+                :date-bound {:future "Framtida"
+                              :none "Ingen gräns"
+                              :past "Tidigare"}
```
Comment (Macroz) [inline resources/translations/sv.edn:283]: Once more!

File: src/clj/rems/api/schema.clj [hunk 4]
```diff
@@ -134,6 +134,9 @@
   {:field/id schema-base/FieldId
    :field/type (s/enum :attachment :date :description :email :header :ip-address :label :multiselect :option :phone-number :text :texta :table)
    :field/title schema-base/LocalizedString
+   (s/optional-key :field/date-bound) (rjs/field
+                                         (s/enum :none :past :future)
```
Comment (Macroz) [inline src/clj/rems/api/schema.clj:138]: Should we skip `:none` and consider a missing `field/date-bound` to be the same as it? If the bound does not exist, it does not need the value?

This follows the logic we have for `field/privacy` and `field/visibility` though. But perhaps they are sort of wrong too.

**Review by Macroz — APPROVED** [review]
Thanks, looking good! I can see only minor improvements that you can fix if you like.

We'll have a manual click-through / we can test this in our regular demo meeting and take care of any future improvements ourselves. But I'll get back to you once we have done that so you can decide if you want to do them or shall we.

I'll leave this unmerged until we have done that and have possibly more feedback.

**Comment by Macroz** [comment]
`cljfmt` fails because of some other indentation issue. You can run `lein cljfmt check` locally.

Also the unit test that fails is saying one of the localizations `:t.actions.errors/date-not-in-future :t.actions.errors/date-not-in-past` is perhaps typoed or missing somewhere?

**Comment by Macroz** [comment]
We got an error from missing the `time` when trying to set a bound for an existing form date field (example form of all fields types).

**Comment by ossilva** [comment]
Thanks for the review! I created this branch to track progress in favour of the deprecated PR --  happy to hear some parts are already looking ready.

I'll try to reproduce the missing `time` error and wrap things up on my end this weekend.

**Comment by Macroz** [comment]
Hey @ossilva,

Do you still want to keep this PR and perhaps try to finish it or shall we adopt it and finish it to get it merged?

**Comment by Macroz** [comment]
I'm cleaning up the pull requests, so I'll close this one since it needs some updates. Please reopen/send again, if you want to continue!

**Pull request closed without merging**